packages/icon/src/isComponentGlyph.ts

```diff
@@ -2,19 +2,23 @@ import { ComponentType, isValidElement, ReactNode } from 'react';
 
 import { LGGlyph } from './types';
 
-type ExtendedComponentType = ComponentType<any> & {
-  [key: string]: any;
-};
+type ExtendedComponentType =
+  | ComponentType<React.PropsWithChildren<any>> & {
+      [key: string]: any;
+    };
 /**
  * Helper type to check if element is a LeafyGreen UI Glyph
  * @internal
  */
+function isComponentGlyph<P>(
+  node: {} | null | undefined,
+): node is React.ReactElement<P>;
 function isComponentGlyph(node: ReactNode): node is LGGlyph.Element;
 function isComponentGlyph(
   component: ExtendedComponentType,
 ): component is LGGlyph.Component;
 function isComponentGlyph(
-  child: ReactNode | ExtendedComponentType,
+  child: ReactNode | ExtendedComponentType | unknown,
```

Comment:
@bruugey im curious why we needed to add this? It created a few ts issues with `isComponentGlyph` in sidenav and combobox

Reply:
there's gotta be a better way to do what i did here, but basically we're passing something with type unknown here in Tooltip - i have a slightly different fix running locally, but there are a couple of different ways we can go about this.

Reply:
either we can add another level of checking in Tooltip before calling this function (maybe preferred) or add stronger typing for supporting an unknown coming in (which is what the validate call is doing from inside the function)

Reply:
Fixed this!